scripts/build_climate_projections.py

```diff
@@ -0,0 +1,210 @@
+# -*- coding: utf-8 -*-
+# SPDX-FileCopyrightText:  PyPSA-Earth and PyPSA-Eur Authors
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+# -*- coding: utf-8 -*-
+"""
+Creates a dataset corresponding to the projected climate in a requested year.
+
+Relevant Settings
+-----------------
+
+.. code:: yaml
+
+    projection:
+        climate_scenario:
+        present_year:
+        future_year:
+        years_window:
+        gcm_selection:
+
+Inputs
+------
+
+- ``cutouts/cutout.nc``: confer :ref:`cutout`, a cutout file produced by altile
+- ``data/cmip6_avr.cn``:
+
+Outputs
+-------
+
+- ``cutouts/{cutout}_{future_year}.nc"``: A cutout modified to account for future climate conditions
+
+Description
+-----------
+
+The rule :mod:`build_climate_projections` creates a cutout file which corresponds to a requested year in the future. Temperature projectons are being calculated combining data for cutout.nc which is assumed to be representative of the past climate and an ensemble of CMIP6 globale climate models to account for the future climate
+"""
+
+
+import datetime as dt
+import os
+
+import atlite
+import geopandas as gpd
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import pypsa
+import xarray as xr
+from _helpers import configure_logging, create_logger
+from shapely.geometry import LineString as Line
+from shapely.geometry import Point
+
+logger = create_logger(__name__)
+
+
+def crop_cmip6(cmip6_xr, cutout_xr):
+    # spartial margin needed to avoid data losses during further interpolation
+    d_pad = 1
+
+    cmip6_region = cmip6_xr.sel(
+        lat=slice(
+            min(cutout_xr.coords["y"].values) - d_pad,
+            max(cutout_xr.coords["y"].values) + d_pad,
+        ),
+        lon=slice(
+            min(cutout_xr.coords["x"].values) - d_pad,
+            max(cutout_xr.coords["x"].values) + d_pad,
+        ),
+    )
+    return cmip6_region
+
+
+def interpolate_cmip6_to_cutout_grid(cmip6_xr, cutout_xr):
+    # TODO read from the cutout file instead of hardcoding
+    dx_new = 0.3
+
+    newlon = np.arange(
```

Comment:
May be good to use np.max and alike but unsure performences increas significantly

Reply:
Thanks a lot for the hint! Definitely worth investigation.

Reply:
Have done some profiling, and `np.min( )`, `np.max( )` have been 5-10% faster. Agree that it's not breakthrough in performance, but it is consistent improvement which doesn't affect code readability. So, I'd opt for it. Thanks for the suggestion! :)

Reply:
Thanks for checking, by looking at the atlite code it raised few ideas:
1. is it possible to load the cmip file as an atlite cutout? this doesn't mean to use the cutout for atlite but simply leverage on the utility functions it has
2. if not, the cutout.py file gives quite some hints to improve the functionalities if the task is actually time consuming. For example, the minimum value may always be the first of the list while the maximum the last one (to verify though)

Link to the cutout.py file : https://github.com/PyPSA/atlite/blob/a71bca2f6f7221b70dbbc371752aef56d937b1ff/atlite/cutout.py#L314